lib/cdo/school_autocomplete.rb

```diff
@@ -1,27 +1,33 @@
 require 'singleton'
-require 'triez'
 
 class SchoolAutocomplete
   include Singleton
 
-  def initialize
-    @schools = Triez.new value_type: :object
-    School.all.each do |school|
-      key = "#{school[:name]} #{school[:city]} #{school[:zip]}".downcase
-      @schools.change_all(:suffix, key) {school}
-    end
-  end
-
   def get_matches(query, limit)
     results = []
-    if query.length >= 3
-      @schools.search_with_prefix(query.downcase, limit: limit) do |_, school|
-        results << Serializer.new(school).attributes
+    if query.length >= 4
+      schools = School.
+        where("MATCH(name,city) AGAINST(? IN BOOLEAN MODE)", to_search_string(query)).
+        limit(limit)
+      results = schools.map do |school|
+        Serializer.new(school).attributes
       end
     end
     return results
   end
 
+  # Formats the query string for boolean full-text search.
+  # For instance, if the user-defined query string is 'abc dec',
+  # the string is reformatted to '+ABC +DEF*'.
+  # See: https://dev.mysql.com/doc/refman/5.6/en/fulltext-boolean.html
+  private def to_search_string(query)
+    words = []
+    query.strip.split(/\s+/).each do |w|
+      words << "+#{w.gsub(/\W/, '').upcase}"
+    end
```

Comment:
It's worth compacting here too, in case one of the "words" consists entirely of non-word characters and ends up empty after this gsub.

Also better to map rather than enumerate and add to an array. Maybe something like this:

```ruby
  words = query.strip.split(/\s+/).map do |word|
      "+#{w.gsub(/\W/, '').upcase}"
  end.map(&:presence).compact
```

[presence](https://apidock.com/rails/Object/presence) will return nil for empty strings, and otherwise return the string itself, and [Array.compact](https://apidock.com/ruby/v1_9_3_392/Array/compact) will remove the nils.